jvector-base/src/main/java/io/github/jbellis/jvector/graph/similarity/BuildScoreProvider.java

```diff
@@ -153,15 +154,15 @@ public SearchScoreProvider diversityProviderFor(int node1) {
                 // like searchProviderFor, this skips reranking; unlike sPF, it uses pqv.scoreFunctionFor
                 // instead of precomputedScoreFunctionFor; since we only perform a few dozen comparisons
                 // during diversity computation, this is cheaper than precomputing a lookup table
-                VectorFloat<?> v1 = vts.createFloatVector(dimension);
+                VectorFloat<?> v1 = reusableVector.get();
```

Comment:
We should probably document here or in the parent interface that the resulting `SearchScoreProvider` is not thread safe. I reviewed the current usages of this result and of the result from `searchProviderFor`, and we never publish the result of the `SearchScoreProvider` to another thread, so it is correct as is.

Reply:
I agree that it should be documented. I will add a comment. Thanks!

Reply:
Thanks!